### DISCUSSION TITLE
Debug/Release configurations set through code behind

### DISCUSSION THREAD

**Discussion by david-d-le** [AdamEssenmacher/MemoryToolkit.Maui#21] [Ideas]
Hi,
I am new to all of this, but I was wondering if it would be possible to set ``mtk:LeakMonitorBehavior.Cascade="True"`` and ``mtk:TearDownBehavior.Cascade="True"`` in code behind in the pages to be able to differentiate between Debug and Release configurations.

I tried this:

```csharp
public MainPage() : ContentPage
{
#if DEBUG
    LeakMonitorBehavior.SetCascade(this, true);
    TearDownBehavior.SetCascade(this, false);
#else
    LeakMonitorBehavior.SetCascade(this, false);
    TearDownBehavior.SetCascade(this, true);
#endif
    InitializeComponent();
}
```

I just need to know if this solution will work same as writting it into the XAML because I would like to have a possibility to easily disable everything in production if problems occurs.

**Comment by AdamEssenmacher** [comment]
This looks right.
